pkg/util/kubernetes/apiserver/apiserver.go

```diff
@@ -141,6 +141,12 @@ func GetAPIClient() (*APIClient, error) {
 	return globalAPIClient, nil
 }
 
+// ReSetGlobalAPIClientOnce resets globalAPIClientOnce
+func ReSetGlobalAPIClientOnce() {
+	log.Info("Reset globalAPIClientOnce to recreate APIClient")
+	globalAPIClientOnce = new(sync.Once)
```

Comment:
This is abuse of sync.Once, and a data race if GetAPIClient happens to run concurrently. (although it will probably work in practice).

If the orchestrator check modifies the state of the API client in an irreversible way, perhaps it should use a dedicated private instance instead of a shared one.

What happens to informers in other running checks when we reset the client like this? (I see helm check also uses the informer API).

Reply:
Thank you for the review and it's very helpful. I checked helm check and it uses [a private informer factory,](https://github.com/DataDog/datadog-agent/blob/main/pkg/collector/corechecks/cluster/helm/helm.go#L201) which should be a good solution for orchestration check as well. [ef839bd](https://github.com/DataDog/datadog-agent/pull/15616/commits/ef839bd0e1756b9549598a15e6834d24770af2b5)